irods/path/__init__.py

```diff
@@ -9,20 +9,39 @@ class iRODSPath(str):
 
     def __new__(cls, *elem_list, **kw):
         """
-        Initializes our immutable 'str' object with a normalized form.
+        Initializes our immutable string object with a normalized form.
+        An instance of iRODSPath is also a `str'
 
-        Keywords may include 'absolute'. The default is True, forcing a slash as
-        the leading character of the resulting string object.
+        Keywords may include only 'absolute'. The default is True, forcing a slash as
+        the leading character of the resulting string.
 
         Variadic parameters are the path elements, strings which may name individual
-        collections or sub-hierarchies (internally slash-separated):
+        collections or sub-hierarchies (internally slash-separated).  These are then
+        joined using the path separator:
 
-        In the resulting object, trailing and redundant path separator are removed,
-        as is the "trivial" path element ('.'):
+          data_path = iRODSPath( 'myZone', 'home/user', './dir', 'mydata')
+          # => '/myZone/home/user/dir/mydata'
 
-        c = iRODSPath('/tempZone//home',username)
-        session.collections.get( c )
+        In the resulting object, any trailing and redundant path separators are removed,
+        as is the "trivial" path element ('.'), so this will work:
 
+            c = iRODSPath('/tempZone//home/./',username + '/')
+            session.collections.get( c )
+
+        If the `absolute' keyword hint is set to False, leading '..' elements are not
+        suppressed (since only for absolute paths is "/.." equivalent to "/"), and the
+        leading slash requirement will not be imposed on the resulting string.
+        Note also that a leading slash in the first argument will be preserved regardless
+        of the `absolute' hint, but subsequent arguments will act as relative paths
+        regardless of leading slashes. So this will do the "right thing":
+
+            my_dir = str(iRODSPath('dir1'))                     # => "/dir1"
+            my_rel = ""+iRODSPath('dir2', absolute=False)       # => "dir2"
+            my_abs = iRODSPath('/Z/home/user', my_dir, my_rel)  # => "/Z/home/user/dir1/dir2"
+
+        Finally, because iRODSPath has class `str` as a base class, this is also possible:
+
+            iRODSPath( '//zone/home/public/this', iRODSPath('../that',absolute=False))
```

Comment:
please show the result, just in case someone can't 'see' it.

Reply:
Done...